partridge/gtfs.py

```diff
@@ -147,10 +155,44 @@ def _prune(self, filename: str, df: pd.DataFrame) -> pd.DataFrame:
                 depcol = deps[depfile]
                 # If applicable, prune this dataframe by the other
                 if col in df.columns and depcol in depdf.columns:
-                    df = df[df[col].isin(depdf[depcol])]
+                    converter = self._get_convert_function(filename, col)
+                    # Convert the column before pruning since depdf is already converted
+                    col_series = converter(df[col]) if converter else df[col]
+                    df = df[col_series.isin(depdf[depcol])]
 
         return df
 
+    def _get_convert_function(self, filename, colname):
+        """return the convert function from the config
+        for a specific file and column"""
+        return self._config.nodes.get(filename, {}).get("converters", {}).get(colname)
+
+    def _validate_dependencies_conversion(self):
+        """Validate that dependent columns in different files
+        has the same convert function if one exist.
+        """
+
+        def check_column_pair(column_pair: dict) -> bool:
+            assert len(column_pair) == 2
+            convert_funcs = [
+                self._get_convert_function(filename, colname)
+                for filename, colname in column_pair.items()
+            ]
+            if convert_funcs[0] != convert_funcs[1]:
+                return False
+            return True
+
+        for file_a, file_b, data in self._config.edges(data=True):
+            dependencies = data.get("dependencies", [])
+            for column_pair in dependencies:
+                if check_column_pair(column_pair):
+                    continue
+                warn(
```

Comment:
Why produce a warning here as opposed to raising an exception?

Reply:
I thought that it might be intentional, for example int8 and int16, or something like that.